master/internal/rm/multirm/multirm_intg_test.go

```diff
@@ -680,3 +680,72 @@ func TestDisableSlot(t *testing.T) {
 	require.Equal(t, err, ErrRMNotDefined("bogus"))
 	require.Empty(t, ret)
 }
+
+func TestGetRMName(t *testing.T) {
+	def := mocks.ResourceManager{}
+	def.On("GetResourcePools").Return(&apiv1.GetResourcePoolsResponse{}, nil)
+
+	gcp := mocks.ResourceManager{}
+	gcp.On("GetResourcePools").Return(&apiv1.GetResourcePoolsResponse{
+		ResourcePools: []*resourcepoolv1.ResourcePool{
+			{Name: "gcp1"}, {Name: "gcp2"},
+		},
+	}, nil)
+
+	aws := mocks.ResourceManager{}
+	aws.On("GetResourcePools").Return(&apiv1.GetResourcePoolsResponse{
+		ResourcePools: []*resourcepoolv1.ResourcePool{
+			{Name: "aws1"}, {Name: "gcp2"},
+		},
+	}, nil)
+
+	mockMultiRM := MultiRMRouter{
+		defaultRMName: "default",
+		rms: map[string]rm.ResourceManager{
+			"default": &def,
+			"gcp":     &gcp,
+			"aws":     &aws,
+		},
+		syslog: logrus.WithField("component", "resource-router"),
+	}
+
+	// If the RM/RP are not named, return default RM name.
+	rmName, err := mockMultiRM.getRM("", "")
+	require.Nil(t, err)
+	require.Equal(t, mockMultiRM.defaultRMName, rmName)
+
+	// If the RP isn't named, return the explicitly named RM, if it exists.
+	rmName, err = mockMultiRM.getRM("aws", "")
+	require.Nil(t, err)
+	require.Equal(t, "aws", rmName)
```

Comment:
nit: can turn these into subtests

or table driven tests

Reply:
for sure, will wait to change until getRM behavior is approved though